sky/spot/recovery_strategy.py

```diff
@@ -305,6 +306,12 @@ def recover(self) -> float:
                 new_resources = resources.copy(cloud=launched_cloud,
                                                region=launched_region)
                 task.set_resources({new_resources})
+
+                # Note: Preempted TPU VM cannot be reused and needs to be
+                # cleaned up. Otherwise, it will occupy the quota.
+                is_tpuvm = tpu_utils.is_tpu_vm(new_resources)
+                if is_tpuvm:
+                    self.terminate_cluster()
```

Comment:
This will only be effective for a managed spot VM. What will happen if the `sky status -r` is called when there is a preempted spot TPU VM in the status table? What status do we show for that cluster?

I am leaning towards having the termination in the `refresh_cluster_status` in backend_utils.py, so that a spot TPU VM launched with `--use-spot` can be handled correctly as well.

Reply:
Looks like we clean up preempted spot TPU VM during `sky status -r`.

https://github.com/skypilot-org/skypilot/blob/1955bee5a337508af67de28615ed03a594fa30cf/sky/backends/backend_utils.py#L1551-L1556

do you mean we should just call `refresh_cluster_status` here instead of `self.terminate_cluster()`?

Reply:
Discussed offline. We decided to move the termination of the cluster into the `refresh_cluster_status`